### DISCUSSION TITLE
Select SID (Service ID) to be streamed as paremeter in RTSP URL (MRL)?

### DISCUSSION THREAD

**Discussion by MadsBakholt** [Barracuda09/SATPI#204] [General]
Hi,

Having installed SatPI in a Proxmox hosted container, I'm having "things running" upon DVB-T2 tuner and a proper antenna - so far, so good! :)

But... having created a channels.m3u for use in e.g. VLC I'm in the situation that I can only state the SID (Service ID AKA "program" AKA "channel") as a #EXTVLCOPT in the m3u file, e.g. "#EXTVLCOPT:program=10005" and not as a parameter in the RTSP URL.

Having read up on the SAT>IP protocol specifications and especially the RTSP part, I can only see the ordinary parameters as "frequency", "msys", etc. and not a "sid" or "program" parameter - not part of the specifications. :-/

Thing is: I cannot just rely on the EXTVLCOPT in the m3u file as this "extension" is ignored by many players, even VLC based players...

So my question is: is there a SatPI specific RTSP parameter to select the SID to stream or similar?

**Comment by Barracuda09** [comment]
Hi @MadsBakholt 

No unfortunately there, is at the moment no way to select a SID with a parameter `sid=`.
So you only want to select the SID and SatPI needs to "find" the correct PIDs for this service?

So this has to go on the TODO list

**Reply by MadsBakholt** [reply]
Correct - since all PIDs are in the PMT along with SID reference, it should be possible (not saying easy)... thing is, of course I can dig out the individual PIDs and hand'em over to VLC (though I think there's a bug in VLC with multiple PIDs) rather than just stating a SID = the SID approach is more user friendly! ;)

**Reply by Barracuda09** [reply]
If you tune to a frequency and give `pids=0,1,16,17,18` and get SatPI.xml like:
http://192.168.1.13:8875/SatPI.xml
You will find in the XML file the `<filter>` element with the PAT, SDT and NIT and if selected a PMT with PID list. So all info is there only the algorithm to select the correct SID and list all PMT in the PAT is missing at the moment.

Edit NIT:
This is not finished for all DVB systems

**Reply by MadsBakholt** [reply]
Ach... I had it running yesterday, perfectly tuning a DVB-T2 channel from the URL:
rtsp://192.168.1.200:554/?freq=626&bw=8&gi=14&msys=dvbt2&mtype=256qam&fec=23&pids=all

That doesn't work today... to be honest, I've switched forth and back between MiniSatIP and SatPI - cannot remember what I did on SatPI to get VLC to stream the TS stream! :-/

**Reply by MadsBakholt** [reply]
But no <filter> element - I wonder if I'm running on latest SatPI - checking!

**Reply by MadsBakholt** [reply]
It IS the latetst and greatest SatPI (freshly compiled) - God damnit, wonder why it's not working today in VLC! :/
The URL above looks correct?

**Reply by Barracuda09** [reply]
Could you give the log when doing the request with VLC?  you can activate in config page `Log messages to syslog`

**Reply by MadsBakholt** [reply]
Never mind... it totally works - switched to another laptop with Linux installed in conjunction with VLC! :-)

But... VLC have a bug and I think it's wellknown - if you state pids=1012,1022,1032 (vid, aid and tid) = no audio, only video!

But collaborate a bit more on the <filter> element? If I tune in on the before mentioned RTSP URL with pids=all my <filter> element looks like:
```
<filter>
<pidcsv>all</pidcsv>
<totalCCErrors>43</totalCCErrors>
<filterPCR>
<inputtype>checkbox</inputtype>
<value>false</value>
</filterPCR>
<addUserPids>
<inputtype>text</inputtype>
<value/>
</addUserPids>
<channelname>Not Found</channelname>
<networkname>Not Found</networkname>
<pat/>
<pmtlist/>
<sdt>
<transportStreamID>0</transportStreamID>
<networkID>0</networkID>
</sdt>
<nit>
<networkID>0000</networkID>
<networkName/>
</nit>
</filter>
```

So is it only filled in when I specify PIDs in the URL or?

**Reply by Barracuda09** [reply]
It could be, you could try only adding `pids=0,1,16,17,18` and it has to collect the data, so it needs some time to populate this.
